models/health.go

```diff
@@ -87,18 +94,69 @@ type ProxyStatus struct {
 // - Inbound//Outbound are the rates of requests by protocol and status_code.
 //   Example:   Inbound: { "http": {"200": 1.5, "400": 2.3}, "grpc": {"1": 1.2} }
 type RequestHealth struct {
-	Inbound  map[string]map[string]float64 `json:"inbound"`
-	Outbound map[string]map[string]float64 `json:"outbound"`
+	Inbound            map[string]map[string]float64 `json:"inbound"`
+	Outbound           map[string]map[string]float64 `json:"outbound"`
+	inboundSource      map[string]map[string]float64
+	inboundDestination map[string]map[string]float64
 }
 
 // AggregateInbound adds the provided metric sample to internal inbound counters and updates error ratios
 func (in *RequestHealth) AggregateInbound(sample *model.Sample) {
-	aggregate(sample, in.Inbound)
+	// Samples need to be aggregated by source or destination reporter, but not accumulated both
+	reporter := string(sample.Metric[model.LabelName("reporter")])
+	switch reporter {
+	case "source":
+		aggregate(sample, in.inboundSource)
+	case "destination":
+		aggregate(sample, in.inboundDestination)
+	default:
+		log.Tracef("Inbound metric without reporter %v ", sample)
+		aggregate(sample, in.Inbound)
+	}
 }
 
 // AggregateOutbound adds the provided metric sample to internal outbound counters and updates error ratios
 func (in *RequestHealth) AggregateOutbound(sample *model.Sample) {
-	aggregate(sample, in.Outbound)
+	// Outbound traffic will be aggregated per source reporter
```

Comment:
I am annoyed by the absence of symmetry between how inbound and outbound metrics are processed. Inbound and outbound are really the same metrics, just the point of view differs, so if we had an issue with the inbound metrics, I would imagine the same issue exists with the outbound metrics

Reply:
Again, I'll try to detach the global discussion about reporters, but narrow that on health endpoint only.
Otherwise we won't progress.

Today, Outbound metrics are reported used source, but Inbound metrics use both.
You can see this in graph sidepanels and traffic tabs.

The goal of this fix was to address the health endpoint, in that we were aggregating both, that's a problem.

What to use ?

Well, for Outbound metrics is a consensus to use the source reporter (I don't enter in the discussion if that's good or bad, yet).

For Inbound metrics, there are more logic, in the "heatlh" context as results may differ I think using the max could help to spot a problem.

I could apply same logic in the Outbound as well, but I don't think we can't duplicate same logic in graph.

To me is to choose between:
a) Fixed reporter
b) Simple logic to choose on Inbound, Outbound or both.

As commented, this problerm is more visible on scenarios which errors, where the users see one thing in one reporter and a different one in another.

On NO ERRORS, probably it's identical info, in scenarios with ERRORS I think it could be interesting to show in the healh that an element have more errors (which probably will be more close to the reality).

I insist, but this PR is not a generic reporter source/destination, it should be tied to the health enpoint.

To change the reporter in graph, traffic or any other area perhaps is a deeper task that I didn't want to tackle first.